pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -131,13 +152,15 @@ rec {
     nativeVersion = "0.22-milestone-24";
     sha256 = "0jwycvzv8a5v2bhg5d8zccr2csr3sf9y5mrr9d2ap44p09a8r9kv";
     defaultJava = jdk17;
+    runToolchainTest = true;
```

Comment:
Maybe this could be replaced by a `hasToolchainSupport` that:

- If true, enables the test for toolchains
- If false, prevents users from setting `javaToolchains` to anything other than an empty list

Reply:
This has been done, using a warning instead of a full-on evaluation failure